history.go

```diff
@@ -0,0 +1,88 @@
+package merkletree
+
+import (
+	"errors"
+
+	"github.com/coniks-sys/libmerkleprefixtree-go/crypto"
+)
+
+var (
+	ErrBadEpoch           = errors.New("[merkletree] Next epoch's STR has bad epoch")
+	ErrSTRNotFound        = errors.New("[merkletree] STR not found")
+	ErrMalformedHashChain = errors.New("[merkletree] Malformed hash chain")
+)
+
+type History struct {
+	currentSTR    *SignedTreeRoot
+	hashChain     map[int64]*SignedTreeRoot
```

Comment:
Not sure: Wouldn't it be nice to define a type which aliases `int64` and call it `epoch` (or sth. similar to `type Epoch int`) . So everyone sees the map definition, directly recognizes, that this is in fact a map from epochs to signed tree roots?